### ISSUE TITLE
Japanese/Chinese gets mixed up in the Slug field

### ISSUE THREAD

**Issue by CptSupermrkt** [jazzband/django-taggit#531]
Heya, I'm using a Python blog project (Puput) which uses taggit for the blog tags.  I noticed that when I enter a blog post with tags that contain Japanese, the taggit "slug" ends up being translated to English-ified Chinese.  For example, I entered a Japanese tag, 甲子園.  This Japanese is read in English as "koushien".  However, taggit enters this into the database:

```
sqlite> select * from taggit_tag;
id|name|slug
1|甲子園|jia-zi-yuan
```

jia-zi-yuan is Chinese, not Japanese.

I looked into this a bit, and it seems the actual core problem here is related to the built-in Django slugify function.  This is recreatable at that level (before taggit is even involved):

```
>>> from django.utils.text import slugify
>>> from unidecode import unidecode
>>> slugify(unidecode("甲子園"))
'jia-zi-yuan'
```

It seems, though, that as of Django 1.9, the slugify function takes "allow_unicode=True" to allow slugs to contain the native Asian characters.  This seems ideal:

```
>>> slugify("甲子園", allow_unicode=True)
'甲子園'
```

It would be nice if Taggit allowed for customization of this --- how tag slugs are created, with allow_unicode on or off.

I realize in the docs there is a section about how to override slug logic, and that's great, that's probably what I'll end up doing.  But still, because the default behavior ends up giving results in the wrong language, I'd classify this as a bug that can be improved upon, if only by adding a settings.py option like TAGGIT_ALLOW_SLUG_UNICODE, etc.

**Comment by jdufresne** [comment]
> jia-zi-yuan is Chinese, not Japanese.

This looks like an issue with the upstream Unidecode library. It is documented somewhere here: https://github.com/avian2/unidecode/issues/16

**Comment by rtpg** [comment]
I think that allowing unicode slugs would be very good (tho at what point are slugs no longer slugs?), I think that how slugs work tends to have enough issues that we should help people out.

(By the way you can redefine the slugification function to avoid this kind of issue in your own case)

**Comment by rtpg** [comment]
Alright this has been fixed by, by default, not using `unidecode`, and instead storing the unicode characters as slugs by default. This is available from django-taggit 3.0.0
